### PR TITLE
Enum for Datasources

### LINKED ISSUE #50: Add ENUM for data sources
Creating a fixed Enum that allows someone to select a source by name to fetch the source id would allow easier calls to API endpoints.

### PR DESCRIPTION
**Pull request by akash5100**
* closes #50

### CODE REVIEW

File: hvpy/datasources.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+from enum import Enum
+
+
+class datasources(Enum):
```
Comment (nabobalis) [inline hvpy/datasources.py:4]:
```suggestion
class DataSources(Enum):
```

File: hvpy/datasources.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
+from enum import Enum
+
+
```
Comment (nabobalis) [inline hvpy/datasources.py:3]:
```suggestion

__all__ = ["DataSources"]
```

**Comment by nabobalis** [comment]
I think a new AI page needs to be created for this, or this enum needs to be imported into another init file. Maybe the highest level one.

**Comment by nabobalis** [comment]
Also a basic test that imports the enum and checks it works.

**Comment by akash5100** [comment]
AI page?

**Comment by nabobalis** [comment]
Typo, I meant API

**Review by nabobalis — APPROVED** [review]

File: hvpy/datasources.py [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+    EUI_HRI_174 = 86
+    """
+    Solar Orbiter instrument HRI, measurement 174.
+    """
+
+    EUI_HRI_1216 = 87
+    """
+    Solar Orbiter instrument HRI, measurement 1216.
+    """
+
```
Comment (dgarciabriseno) [inline hvpy/datasources.py:440]: A few more to add that are much more recent

|    88 | GOES-R SUVI 94                  |
|    89 | GOES-R SUVI 131                 |
|    90 | GOES-R SUVI 171                 |
|    91 | GOES-R SUVI 195                 |
|    92 | GOES-R SUVI 284                 |
|    93 | GOES-R SUVI 304                 |

File: hvpy/tests/test_datasources.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+from hvpy.datasources import DataSources
+
+
+def test_datasources():
```
Comment (dgarciabriseno) [inline hvpy/tests/test_datasources.py:4]: Add a test where you query getDataSources endpoint and confirm that each ID returned is in the enum.

File: hvpy/tests/test_datasources.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    enum_value_list = [s_id.value for s_id in DataSources]
+    res = getDataSources()
+
+    find_key(res, "sourceId")
+
+    for sid in sids:
+        assert sid in enum_value_list
```
Comment (nabobalis) [inline hvpy/tests/test_datasources.py:24]:
```suggestion
    enum_values = [s_id.value for s_id in DataSources]
    datasources = getDataSources()
    find_key(datasources, "sourceId")
    for source_id in source_ids:
        assert source_id in enum_values
```

File: hvpy/tests/test_datasources.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+def find_key(obj, key):
+    if key in obj:
+        sids.append(obj[key])
+    else:
+        for k, v in obj.items():
+            if isinstance(v, dict):
+                find_key(v, key)
+            else:
+                continue
```
Comment (nabobalis) [inline hvpy/tests/test_datasources.py:14]:
```suggestion
```

File: hvpy/tests/test_datasources.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+def test_datasources():
+    source_id_list = list()
+    enum_value_list = [s_id.value for s_id in DataSources]
+
+    response = getDataSources()
+
+    find_nested_keys(response, "sourceId", source_id_list)
+
+    for source_id in source_id_list:
+        assert source_id in enum_value_list
```
Comment (nabobalis) [inline hvpy/tests/test_datasources.py:34]:
```suggestion
    source_ids = []
    enum_values = [s_id.value for s_id in DataSources]
    response = getDataSources()
    find_nested_keys(response, "sourceId", source_ids)
    for source_id in source_ids:
        assert source_id in enum_values
```

**Review by nabobalis — APPROVED** [review]

**Pull request merged**